prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -150,43 +153,61 @@ class ActionHistory(
         action.username = userName
     }
 
-    fun filterParameters(request: HttpRequestMessage<String?>): String {
-        // list of values to search and filter
-        val notAllowedHeaderParts = listOf("key", "cookie", "auth")
-        val notAllowedParameterParts = listOf("code")
+    /**
+     * Class captures data about the HTTP request from a sender to the report endpoint
+     *
+     */
+    data class ReceivedReportSenderParameters(
+        val method: HttpMethod,
+        val url: String,
+        @JsonProperty("Headers")
+        val headers: Map<String, String>,
+        @JsonProperty("QueryParameters")
+        val queryParameters: Map<String, String>,
+    ) {
+        companion object {
+            // These header/query params can potentially contain auth information, and we do not want to log them
+            // gitleaks:allow
+            private val parameterNamesToExclude = listOf(
+                "key",
+                "cookie",
+                "auth", // gitleaks:allow
+                "code" // gitleaks:allow
+            )
 
-        // Extract query parameters
-        val filteredParams = request.queryParameters.entries
-            .filter { (key, _) ->
-                notAllowedParameterParts.none { part ->
-                    key.contains(part, ignoreCase = true)
+            fun removeExcludedParameters(it: Map.Entry<String, String>) =
+                !parameterNamesToExclude.any { excluded ->
+                    it.key.contains(
```

Comment:
We could mistakenly be filtering out any header that contains the substring listed out in `parameterNamesToExclude`.

Example: `WWW-Authenticate` is gone because it contains `auth`.

Reply:
I agree...but I wanted to add the underlying logic the same